pkg/parse/namespace.go

```diff
@@ -205,30 +220,45 @@ func (p *namespace) setRenderingStatusWithRetires(ctx context.Context, newStatus
 		return status.APIServerError(err, "failed to get RepoSync for parser")
 	}
 
-	if rs.Status.Rendering.Commit != newStatus.commit {
+	currentRS := rs.DeepCopy()
```

Comment:
Looks like we are repeating this same snippet with minor variation a number of times. Might be worth considering an interface for setting the status of each component type so that we could generalize this to a common `setStatus` type function. Not necessarily in scope for this change but just offering the suggestion.

WDYT?

Reply:
I considered that but it's the diff between a RootSync and RepoSync object, and we can't add methods to the API structs. So we would have to wrap them with indirection for everything we need to read/write, which sounds tedious...

If we did that tho, it would be useful for a lot of other RootSync vs RepoSync code, like the reconciler impls.